playbooks/roles/ocp-scheduler/tasks/main.yml

```diff
@@ -0,0 +1,121 @@
+      file:
+        path: "{{ scheduler_workdir }}/{{ item }}"
+        state: directory
+        mode: 0755
+      loop: "{{ scheduler_subdir_list }}" 
+  when: scheduler_workdir != "" and scheduler_subdir_list != []
+
+
+- name: Configure a scheduler profile
+  block:
+    - name: Configure the given scheduler profile by patching the scheduler object
+      kubernetes.core.k8s_json_patch:
+        kind: Scheduler
+        namespace: default
+        name: cluster
+        patch:
+          - op: add
+            path: /spec/profile
+            value: "{{ scheduler_profile }}"
+
+    - name: Wait for 5 mins after setting the schedule profile to get necessary pods up in openshift-kube-scheduler project
+      wait_for:
+        timeout: 300
+
+    - name: Create a pods to test scheduler profile
+      kubernetes.core.k8s:
+        state: present
+        definition:
+          apiVersion: v1
+          kind: Pod
+          metadata:
+            name: "{{ item }}"
+            namespace: default
+          spec:
+            containers:
+            - name: demo
+              image: docker.io/ibmcom/pause-ppc64le:3.1
+              imagePullPolicy: IfNotPresent
+              resources:
+                requests:
+                  memory: "{{ scheduler_pod_memory_resource }}"
+                  cpu: "{{ scheduler_pod_cpu_resource }}"
+                limits:
+                  memory: "{{ scheduler_pod_memory_limit }}"
+                  cpu: "{{ scheduler_pod_cpu_limit }}"
+      with_sequence: start=1 end="{{ scheduler_pod_count }}" format=mypod-%d
+    
+    - name: Wait for 5 mins to get the pods in the Running state
+      wait_for:
+        timeout: 300
+    
+    - name: Verify the pods state
+      shell: oc get pods | grep mypod | grep -v Running | wc -l
+      register: pending_pods
+      until: pending_pods.stdout|int == 0
+      retries: 10
+      delay: 30
+
+    - name: Verify the scheduler profile
+      include_tasks: "{{ role_path }}/tasks/verify_scheduler_profile.yaml"
+
+    - name: Delete the created pods
+      shell: oc get pods | awk '/^mypod-/{system("oc delete pod " $1)}'
+
+    - name: Remove the applied profile configuration to get back the default node scheduling behavior
+      kubernetes.core.k8s_json_patch:
+        kind: Scheduler
+        namespace: default
+        name: cluster
+        patch:
+          - op: remove
+            path: /spec/profile
+
+  when: configure_scheduler_profile
+
+- name: Run the e2e test for openshift-kube-scheduler
+  block:
+    - fail:
+        msg: "Please set the environment variables GITHUB_USERNAME and GITHUB_ACCESS_TOKEN"
+      when: github_username == "" and github_personal_access_token == ""
+
+    - name: Clone the git repo
+      git:
+        repo: "https://{{ github_username }}:{{ github_personal_access_token }}@github.com/{{ scheduler_e2e_github_repo | urlsplit('path') }}"
+        dest: "{{ scheduler_workdir }}/openshift-tests-private"
+        version: "master"
```

Comment:
I see this is hardcoded and you are not using `scheduler_e2e_github_branch` variable anywhere.

Reply:
Used the var `scheduler_e2e_github_branch` for assigning a branch name.